File: tools/capture_static.py
import sys
import os
import time
from playwright.sync_api import sync_playwright

def main():
    if len(sys.argv) < 5:
        print("Usage: python capture_static.py <html_file> <output_png> <width> <height>")
        sys.exit(1)
        
    html_file = os.path.abspath(sys.argv[1])
    output_png = os.path.abspath(sys.argv[2])
    width = int(sys.argv[3])
    height = int(sys.argv[4])
    
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page(
            viewport={"width": width, "height": height},
            device_scale_factor=2
        )
        
        file_url = f"file:///{html_file.replace(os.sep, '/')}"
        page.goto(file_url, wait_until="networkidle")
        time.sleep(1) # wait for fonts/setup
        
        page.screenshot(path=output_png)
        browser.close()
        print("✅ Static high-res PNG saved to:", output_png)

if __name__ == "__main__":
    main()

File: tools/capture_static.js
const puppeteer = require('puppeteer');
const path = require('path');

const args = process.argv.slice(2);
if (args.length < 4) {
    console.error("Usage: node capture_static.js <html_file> <output_png> <width> <height>");
    process.exit(1);
}

const HTML_FILE = path.resolve(args[0]);
const OUTPUT_FILE = path.resolve(args[1]);
const WIDTH = parseInt(args[2]);
const HEIGHT = parseInt(args[3]);

(async () => {
    console.log(`Launching browser to capture static high-res image... (${WIDTH}x${HEIGHT})`);
    const browser = await puppeteer.launch({ headless: 'new' });
    const page = await browser.newPage();

    // Use 2x scale for high quality
    await page.setViewport({ width: WIDTH, height: HEIGHT, deviceScaleFactor: 2 });

    const fileUrl = `file:///${HTML_FILE.replace(/\\/g, '/')}`;
    await page.goto(fileUrl, { waitUntil: 'domcontentloaded' });

    // Wait for fonts to load
    await new Promise(r => setTimeout(r, 1000));

    await page.screenshot({ path: OUTPUT_FILE, type: 'png' });

    await browser.close();
    console.log(`✅ Static high-res PNG saved to: ${OUTPUT_FILE}`);
})();
